src/main/java/racingcar/Car.java

```diff
@@ -4,19 +4,41 @@ public class Car {
 
     private static final int MOVE_THRESHOLD = 4;
 
-    private int position = 0;
+    private final CarName name;
+    private Position position;
 
-    public int position() {
+    public Car(CarName name, Position position) {
+        this.name = name;
+        this.position = position;
+    }
+
+    public Car(CarName name) {
+        this(name, new Position());
+    }
+
+    public Car(String name) {
+        this(new CarName(name), new Position());
+    }
+
+    public CarName name() {
+        return name;
+    }
+
+    public Position position() {
         return position;
     }
 
     public void moveIfPossible(int number) {
         if (number >= MOVE_THRESHOLD) {
-            move();
+            position = position.move();
         }
     }
 
-    private void move() {
-        position++;
+    public boolean isAtSamePositionAs(Car maxCar) {
```

Comment:
```suggestion
    public boolean isAtSamePositionAs(Position maxPosition) {
```
Car가 아닌 Position을 전달해 비교하는 것은 어떨까?